packages/Webkul/Checkout/src/Cart.php

```diff
@@ -788,6 +796,40 @@ public function isItemsHaveSufficientQuantity(): bool
         return true;
     }
 
+    /**
+     * Remove cart items, whose product is inactive
+     *
+     * @param \Webkul\Checkout\Models\Cart|null $cart
+     *
+     * @return \Webkul\Checkout\Models\Cart|null
+     */
+    public function removeInactiveItems(CartModel $cart = null): ?CartModel
+    {
+        if (! $cart) {
+            return $cart;
+        }
+
+        foreach ($cart->items as $item) {
+            if ($item->product && $item->product->status === 0) {
+
+                $this->cartItemRepository->delete($item->id);
+
+                if ($cart->items()->get()->count() == 0) {
+                    $this->cartRepository->delete($cart->id);
+
+                    if (session()->has('cart')) {
+                        session()->forget('cart');
+                    }
+                }
+
+                session()->flash('info', __('shop::app.checkout.cart.item.inactive'));
+            }
+        }
+        $cart->save();
```

Comment:
please add a blank line before this line

Reply:
Ok